File: app-ui/src/main/java/com/mercury/platform/ui/frame/setup/scale/SetUpScaleCommander.java
package com.mercury.platform.ui.frame.setup.scale;


import com.mercury.platform.ui.frame.AbstractScalableComponentFrame;
import com.mercury.platform.ui.frame.setup.SetUpCommander;

public class SetUpScaleCommander extends SetUpCommander<AbstractScalableComponentFrame> {
    public SetUpScaleCommander() {
        super();
    }

    @Override
    public void setUpAll() {
        this.frames.forEach((k, v) -> {
            v.setState(ScaleState.ENABLE);
        });
    }

    @Override
    public void setUpAllExclude(Class[] framesClasses) {

    }

    @Override
    public void setOrEndUp(Class frameClass, boolean showingSetUpFrame) {

    }

    @Override
    public void endUpAll() {
        this.frames.forEach((k, v) -> {
            v.setState(ScaleState.DEFAULT);
        });
    }

    @Override
    public void endUp(Class frameClass) {

    }
}

File: app-ui/src/main/java/com/mercury/platform/ui/frame/setup/scale/SetUpScaleCommander.kt
package com.mercury.platform.ui.frame.setup.scale


import com.mercury.platform.ui.frame.AbstractScalableComponentFrame
import com.mercury.platform.ui.frame.setup.SetUpCommander

class SetUpScaleCommander : SetUpCommander<AbstractScalableComponentFrame>() {
    override fun setUpAllExclude(framesClasses: Array<Class<*>>) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun setUpAll() {
        frames.forEach { frame -> frame.setState(ScaleState.ENABLE) }
    }


    override fun setOrEndUp(frameClass: Class<*>, showingSetUpFrame: Boolean) {

    }

    override fun endUpAll() {
        frames.forEach { frame -> frame.setState(ScaleState.DEFAULT) }
    }

    override fun endUp(frameClass: Class<*>) {

    }
}
